adapters/repos/db/shard_accessors.go

```diff
@@ -20,12 +20,44 @@ import (
 	"github.com/weaviate/weaviate/entities/schema"
 )
 
-func (s *Shard) Queue() *VectorIndexQueue {
-	return s.queue
+// ForEachVectorIndex iterates through each vector index initialized in the shard (named and legacy).
+// Iteration stops at the first return of non-nil error.
+func (s *Shard) ForEachVectorIndex(f func(targetVector string, index VectorIndex) error) error {
+	for targetVector, idx := range s.vectorIndexes {
+		if idx == nil {
+			continue
+		}
+
+		if err := f(targetVector, idx); err != nil {
+			return err
+		}
+	}
+	if s.vectorIndex != nil {
+		if err := f("", s.vectorIndex); err != nil {
```

Comment:
question: why are we not using `"default"` as the key for unnamed vector? 
Seems like it'd be in line with the [requirements](https://www.notion.so/weaviate/Requirements-1a070562ccd680c691c8cb006152e51f?pvs=4#1a470562ccd6802bb85efab4ce2f70ba) for this feature.

If there's another document/discussion that addresses that, feel free to answer with just a link.

Reply:
> there could be already customer-created indices named "default"

Seems like that is the _only_ thing that's different about `""` vs. `"default"` though -- that there are no vectors out there named `""`.

Could we then not apply the [same logic](https://github.com/weaviate/weaviate/pull/7606/files#diff-b1e348aef620f02e39f4567eeb5471d0deb1eeeaedbfcba0d6f561e463beb19eR265-R269), but check for an existing named vector first?

```go
if named, ok := namedVectors[name]; ok {
  return named
} else if name == "default" {
  return legacy
}
return nil
```

Reply:
Yeah, that's a viable option. We first need to add validation that a user would not be able to create a "default" vector if legacy is already present.

I will add to project's TODOs to consider this refactoring.

Reply:
Sure 👌 
As I said, this is definitely non-blocking, I just wanted to add some thoughts here